### PR TITLE
PANW Firewall/Panorama add get-pcaps command, panw associates review fixes

### PR DESCRIPTION
**Pull request by bakatzir**
added new get-pcaps commands and its relevant outputs, remove duplica…ted parameter in old commands
FYI - limitation on pcap file is around 10mb, if more only the filename will be downloaded.
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/5840, https://github.com/demisto/etc/issues/16146, https://github.com/demisto/etc/issues/16218

## Screenshots
pcap_is_working

## Does it break backward compatibility?

   - No

## Must have
- [x] Tests - yes, added to the firewall_test_pb
- [x] Documentation - https://github.com/demisto/etc/issues/16102
- [x] Code Review

### CODE REVIEW

**Comment by bakatzir** [comment]
@BenJoParadise  i'll need your advise tomorrow regarding the mocks for this integration.

File: Integrations/integration-Panorama.yml [hunk 1]
```diff
@@ -3357,7 +3352,63 @@ script:
...
+      description: Serial number for the request. For further information, see the
+        Panorama XML API Documentation.
+    - name: searchTime
+      description: Search time for the request. For further information, see the Panorama
+        XML API Documentation.
+    - name: pcapID
+      description: ID of the Pcap for the request. For further information, see the
+        Panorama XML API Documentation.
+    - name: password
+      description: Password for Panorama.
```
Comment (glicht) [inline Integrations/integration-Panorama.yml]: I think the description is wrong. From a quick search I think this is for dlp-pcaps. Please document when this is needed.

File: Integrations/integration-Panorama.yml [hunk 2]
```diff
@@ -3357,7 +3352,63 @@ script:
...
+      required: true
+      auto: PREDEFINED
+      predefined:
+      - application-pcap
+      - filter-pcap
+      - threat-pcap
+      - dlp-pcap
+      description: Type of Packet Capture
+    - name: from
+      description: '"From" value for the rule'
```
Comment (glicht) [inline Integrations/integration-Panorama.yml]: Please provide an example for expected format

File: Integrations/integration-Panorama.yml [hunk 3]
```diff
@@ -3357,7 +3352,63 @@ script:
...
+    - name: to
+      description: '"To" value for the rule'
```
Comment (glicht) [inline Integrations/integration-Panorama.yml]: Please provide an example for expected format

**Comment by bakatzir** [comment]
@glicht 
I forgot to update the args description. fixed.
https://docs.paloaltonetworks.com/pan-os/8-0/pan-os-panorama-api/pan-os-xml-api-request-types/export-files-api/export-packet-captures/export-application-pcaps.html

**Comment by glicht** [comment]
For the `to` parameter what does it mean `Localfile` to download?

**Comment by bakatzir** [comment]
@glicht  you can save an application/dlp pcap with a custom name using this parameter.
@kirbles19  can you help with the description of the `to` argument?
it's here -> https://github.com/demisto/etc/issues/16102

**Comment by kirbles19** [comment]
@bakatzir I think we should change the argument name itself to something more descriptive

- Arg Name: `localName`
- Description: `The new name for the PCAP file after downloading. If this argument is not specified, the file name will be the PCAP file name that was set in the firewall.`

File: Integrations/integration-Panorama.yml [hunk 4]
```diff
@@ -2229,13 +2237,7 @@ script:
...
         result = http_request(URL, 'GET', params = params)
 
-        demisto.results({
-            'Type': entryTypes['note'],
-            'ContentsFormat': formats['json'],
-            'Contents': result,
-            'ReadableContentsFormat': formats['text'],
-            'HumanReadable': 'Get pcap successfully',
-        })
+        demisto.results(fileResult(fromm + ".pcap", result.content))
```
Comment (glicht) [inline Integrations/integration-Panorama.yml]: fromm may be None. Use a default based on current date if missing

File: Integrations/integration-Panorama.yml [hunk 5]
```diff
@@ -2214,28 +2222,29 @@ script:
...
 
         fromm = demisto.args().get('from')
-        to = demisto.args().get('to')
+        localName = demisto.args().get('localName')
         serialno = demisto.args().get('serialNo')
         search_time = demisto.args().get('searchTime')
 
+        file_name = None
         if (fromm):
             params['from'] = fromm
```
Comment (yuvalbenshalom) [inline Integrations/integration-Panorama.yml]: not sure if I remember, is this a typo?

**Comment by bakatzir** [comment]
@yuvalbenshalom 
# panorama-get-pcap: download failed message - @kirbles19  - could you comment here and rephrase this error message?
fw_pcap_fail

# panorama-list-pcaps
Screen Shot 2019-03-20 at 14 58 23

**Comment by yuvalbenshalom** [comment]
looks good, just wait for @kirbles19 to review the error msg

**Comment by bakatzir** [comment]
@yuvalbenshalom  pcap size error message rephrased.

**Pull request merged**